src/main/java/com/hierynomus/smbj/share/DiskShare.java

```diff
@@ -466,6 +468,127 @@ private boolean exists(String path, EnumSet<SMB2CreateOptions> createOptions) th
         }
     }
 
+    /***
+     * Rename(Move) Directory/File. Default will using the false for replaceIfExist option
+     * @param path the path to the original directory/file
+     * @param newFileName the new directory/file name/path after rename
+     * @throws TransportException
+     * @throws SMBApiException
+     */
+    public void rename(String path, String newFileName) throws TransportException, SMBApiException {
```

Comment:
Ok, actually I agree putting the `rename` on both `Directory` and `File` is one of the good designs. If a bug or exception occur only with directory but not file, we can easier to modify and fix it. I could made another pull request for the rename implement on both `File` and `Directory`.

> And yes maybe we should move some more methods also while we're at it. I've been meaning to refactor that part anyway.

I am not sure I got what you mean, do you mean you are going move those `getFileInformation` methods to `File` and `Directory` and add mpre methods to them?